### PR TITLE
Reduce yarn plugin bundle size

### PR DESCRIPTION
**Pull request by mtlewis**
## Hey, I just made a Pull Request!

When combined with the changes in https://github.com/backstage/backstage/pull/27768, these changes reduce the yarn plugin bundle size from 884kb to 12kb(!).

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
🤏 👌

File: packages/yarn-plugin/src/handlers/beforeWorkspacePacking.ts [hunk 1]
```diff
@@ -76,9 +75,8 @@ export const beforeWorkspacePacking = async (
   }
 
   if (
-    some(
-      ['dependencies', 'devDependencies', 'optionalDependencies'],
-      dependencyType => some(rawManifest[dependencyType], hasBackstageVersion),
+    ['dependencies', 'devDependencies', 'optionalDependencies'].some(
+      dependencyType => rawManifest[dependencyType]?.some(hasBackstageVersion),
```
[the remark this answers is not in the document]
Comment (mtlewis) [reply packages/yarn-plugin/src/handlers/beforeWorkspacePacking.ts:79]: Great callout! On second look I agree, this looks like it's now missing some Object.values action, but it's quite curious that the tests (and in particular [this test](https://github.com/backstage/backstage/blob/master/packages/yarn-plugin/src/handlers/beforeWorkspacePacking.test.ts#L77-L99)) pass. I'm gonna investigate that first and fix either this code, or the tests... or maybe both! 😅 . Thanks @freben

[the remark this answers is not in the document]
Comment (mtlewis) [reply packages/yarn-plugin/src/handlers/beforeWorkspacePacking.ts:79]: Ah never mind, the tests actually _don't_ pass, as I would have hoped. I was distracted by the failing E2Es and didn't notice! I'll get this sorted.

[the remark this answers is not in the document]
Comment (mtlewis) [reply packages/yarn-plugin/src/handlers/beforeWorkspacePacking.ts]: Thanks again - this was fixed in 124a108c36ffe. While investigating this I also spotted an opportunity to be more restrictive about typing of the `hooks` object passed to the plugin, added here in f9c9c3ac0723f.

**Pull request merged**